pubsub/cloud-client/quickstart/pub.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START pubsub_quickstart_pub_all]
+import argparse
+import time
+# [START pubsub_quickstart_pub_deps]
+from google.cloud import pubsub_v1
+# [END pubsub_quickstart_pub_deps]
+
+
+def get_callback(api_future, data):
+    """Wraps message data in the context of the callback function."""
+    def callback(api_future):
+        if api_future.exception():
+            print("There was a problem with message {}".format(data))
+        else:
+            print("Published message {} now has message ID {}".format(
+                data, api_future.result()))
+    return callback
+
+
+def pub(project_id, topic_name):
+    """Publishes a message to a Pub/Sub topic."""
+    # [START pubsub_quickstart_pub_client]
+    # Initializes the Publisher client
+    client = pubsub_v1.PublisherClient()
+    # [END pubsub_quickstart_pub_client]
+    # Creates a fully qualified identifier in the form of
+    # `projects/{project_id}/topics/{topic_name}`
+    topic_path = client.topic_path(project_id, topic_name)
+
+    # Data sent to Cloud Pub/Sub must be a bytestring
+    data = "Hello, World!"
+    data = data.encode('utf-8')
+
+    # When you publish a message, the client returns a future.
+    api_future = client.publish(topic_path, data=data)
+    api_future.add_done_callback(get_callback(api_future, data))
+
+    # Keeps the main thread from exiting to handle message processing
+    # in the background.
+    while True:
+        time.sleep(60)
```

Comment:
I think this should be something like while api_future.running(): sleep (0.1) -- this will prevent the code from hanging any longer than necessary.

Reply:
This is much better. I'm removing `while true`.